src/Components/Components/src/PersistentComponentState.cs

```diff
@@ -60,19 +99,26 @@ public PersistingComponentStateSubscription RegisterOnPersisting(Func<Task> call
     [RequiresUnreferencedCode("JSON serialization and deserialization might require types that cannot be statically analyzed.")]
     public void PersistAsJson<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string key, TValue instance)
     {
+        var currentState = CurrentSerializationMode switch
+        {
+            PersistedStateSerializationMode.Server => _currentServerState,
+            PersistedStateSerializationMode.WebAssembly => _currentWebAssemblyState,
+            _ => throw new InvalidOperationException("Invalid persistence mode.")
```

Comment:
How does this work with AutoMode? One of the key points of the design was that calls to `PersistAsJson` still work without specifying the "serialization mode" when the callback comes from a component.

How does that work in this scenario? It's throwing isn't it?

Reply:
Haven't tried this scenario yet.